### ISSUE TITLE
💅 `useExplicitFunctionReturnType` should not be run on non-TypeScript files

### ISSUE THREAD

**Issue by Conaclos** [biomejs/biome#4089]
### Environment information

```bash
Main
```

### Rule name

useExplicitFunctionReturnType

### Playground link

https://biomejs.dev/playground/?lintRules=all&code=ZgB1AG4AYwB0AGkAbwBuACAAZgAoACkAIAB7AH0A

### Expected result

We should run `useExplicitFunctionReturnType` only on TypeScript-enabled files.
We could even exclude declaration files.

@kaykdm could be interested in this issue.

See the implementation of `noEvolvingTypes` that only run on TypeScript enabled files.

### Code of Conduct

- [X] I agree to follow Biome's Code of Conduct
